plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -412,10 +412,21 @@ public void setDatabaseOwner(String databaseName, HivePrincipal principal)
     @Override
     public List<TableInfo> getTables(String databaseName)
     {
-        return glueCache.getTables(databaseName, cacheTable -> getTablesInternal(cacheTable, databaseName));
+        return glueCache.getTables(databaseName, cacheTable -> getTablesInternal(cacheTable, databaseName, _ -> true));
     }
 
-    private List<TableInfo> getTablesInternal(Consumer<Table> cacheTable, String databaseName)
+    @Override
+    public List<String> getTablesWithParameter(String databaseName, String parameterKey, Set<String> parameterValues)
+    {
+        return getTablesInternal(
+                _ -> {},
+                databaseName,
+                table -> table.parameters() != null && parameterValues.contains(table.parameters().get(parameterKey))).stream()
```

Comment:
Do we need to handle the case where `table.parameters().get(parameterKey)` is null separately ? That potentially is nullptr exception